src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -3609,12 +3631,18 @@ void M2(object o)
                 // (8,13): error CS8310: Operator '+' cannot be applied to operand 'default'
                 //         _ = 1 + default!; // 4
                 Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "1 + default!").WithArguments("+", "default").WithLocation(8, 13),
-                // (9,13): error CS8310: Operator '??' cannot be applied to operand 'default'
+                // (9,13): error CS8716: No target type was found for the default literal.
                 //         _ = default ?? d; // 5
-                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default ?? d").WithArguments("??", "default").WithLocation(9, 13),
-                // (10,13): error CS8310: Operator '??' cannot be applied to operand 'default'
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(9, 13),
+                // (10,13): error CS8716: No target type was found for the default literal.
                 //         _ = default! ?? d; // 6
-                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default! ?? d").WithArguments("??", "default").WithLocation(10, 13)
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(10, 13),
+                // (16,13): error CS0019: Operator '==' cannot be applied to operands of type 'object' and 'default'
```

Comment:
>default [](start = 108, length = 7)

This seems weird because 'default' is not a type.

Reply:
I agree the diagnostic isn't great, but it's an existing problem and also occurs with `null` literal:
```
                // (5,10): error CS0019: Operator '??' cannot be applied to operands of type '<null>' and '<null>'
                //         (null ?? null).ToString();
                Diagnostic(ErrorCode.ERR_BadBinaryOps, "null ?? null").WithArguments("??", "<null>", "<null>").WithLocation(5, 10),
               
```
I prefer to leave this as-is for this PR.

---
In reply to: [314092173](https://github.com/dotnet/roslyn/pull/37596#discussion_r314092173) [](ancestors = 314092173)